cpputil/unstructured/tests/regression/fvm_nabla/CMakeLists.txt

```diff
@@ -1,9 +1,15 @@
 add_executable(fvm_nabla driver.cc)
+target_compile_definitions(fvm_nabla PRIVATE STENCIL_IMPL_SOURCE="nabla_naive.hpp")
+add_custom_target(fvm_nabla_stencil DEPENDS nabla_naive.hpp)
```

Comment:
It seems this doesn't work, maybe something like
```set_source_files_properties(driver.cc PROPERTIES OBJECT_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/nabla_naive.hpp)```

For the cuda branch I am not sure if the following works (after gridtools_setup_target) but it's quite a hack:
```set_source_files_properties(driver.cc.cu PROPERTIES OBJECT_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/nabla_cuda.hpp)```

Reply:
I adopted a proper implementation using `configure`. It requires some machinery, but should be solid.